src/groth16/aggregate/prove.rs

```diff
@@ -257,25 +277,31 @@ fn gipa_tipp_mipp<E: Engine>(
 
         // Fiat-Shamir challenge
         // combine both TIPP and MIPP transcript
-        let c_inv = transcript
-            .write(&zab_l)
-            .write(&zab_r)
-            .write(&zc_l)
-            .write(&zc_r)
-            .write(&tab_l.0)
-            .write(&tab_l.1)
-            .write(&tab_r.0)
-            .write(&tab_r.1)
-            .write(&tuc_l.0)
-            .write(&tuc_l.1)
-            .write(&tuc_r.0)
-            .write(&tuc_r.1)
-            .read_challenge();
-
-        // Optimization for multiexponentiation to rescale G2 elements with
-        // 128-bit challenge Swap 'c' and 'c_inv' since can't control bit size
-        // of c_inv
-        let c = c_inv.inverse().unwrap();
+        if first_loop {
+            first_loop = false;
+        // already generated c_inv and c outside of the loop
+        } else {
+            c_inv = *Transcript::<E>::new("gipa")
+                .write(&c_inv)
```

Comment:
to be fair I think we should hash the c and not c_inv - doesnt make much sense to hash the inverse

Reply:
well the paper says to hash the output of the previous hash, which is why I used `c_inv`